test/device-test/CMakeLists.txt

```diff
@@ -125,16 +129,32 @@ foreach(name ${BB02_TEST_LIST})
   # Since noiseprotocol and wallycore are external projects we need to specify the dependency
   add_dependencies(${elf} noise-c)
   add_dependencies(${elf} libwally-core)
-  # We must link against libc before libbitbox because malloc depends on _sbrk
+  # We must link against libc before asf4-drivers-min because malloc depends on _sbrk
   # We must mark "exception_table" as used, otherwise it won't be linked in since it is in a static library
+  # There are circular dependencies between fatfs and bitbox02-platform (therefore group them with linking commands)
   target_link_libraries(${elf}
     PRIVATE
-      c
+      -Wl,-u,exception_table
+      -Wl,--start-group
+      fatfs
       bitbox02-platform
-      asf4-drivers-min
+      -Wl,--end-group
+      samd51a-ds
+      asf4-drivers
       ${QTOUCHLIB_A}
       ${QTOUCHLIB_B}
       ${QTOUCHLIB_T}
+      wallycore
+      secp256k1
+      bitbox02_rust_c
+      noiseprotocol
+      cryptoauthlib
+      base32
+      bignum # TODO: only eth
+      sha3   # TODO: Only eth
+      ctaes
+      c
+      asf4-drivers-min
```

Comment:
These lines seem to be repeated pretty much everywhere - can we store them in a variable? I have the impression we're going to screw up this sequence next time we change the `CMakeLists.txt` and we'll be back to the starting point.

Reply:
yeah that makes sense, as long as it works with the linking order and so on. The rust-library can't be in the variable for example.

Reply:
did a variable.